src/Microsoft.VisualStudio.Threading/SingleThreadedSynchronizationContext.cs

```diff
@@ -83,7 +83,9 @@ public override void Send(SendOrPostCallback d, object state)
             else
             {
                 Exception caughtException = null;
+#pragma warning disable CA2000 // Dispose objects before losing scope
                 var evt = new ManualResetEventSlim();
+#pragma warning restore CA2000 // Dispose objects before losing scope
```

Comment:
📝 This one is a potential bug in microsoft/vs-threading. I'm going to file a follow-up issue for it.

Reply:
Sounds good. I believe disposal is for the native handle that is only allocated when asked for, so it wouldn't be important in this case, but that's an implementation detail and I would support fixing this properly rather than suppressing it.